services/ymax-planner/src/beefy-balance-client.ts

```diff
@@ -0,0 +1,190 @@
+/* eslint-disable max-classes-per-file */
+import { Contract, type WebSocketProvider } from 'ethers';
+import { BeefyPoolPlaces } from '@aglocal/portfolio-contract/src/type-guards.js';
+import {
+  axelarConfig,
+  axelarConfigTestnet,
+} from '@aglocal/portfolio-deploy/src/axelar-configs.js';
+import { Fail, q } from '@endo/errors';
+import type { EvmProviders } from './support.js';
+import type { ClusterName } from './config.js';
+
+type CaipChainId = `eip155:${number}`;
+type HexAddress = `0x${string}`;
+type PoolKey = keyof typeof BeefyPoolPlaces;
+type AccountId = `${string}:${string}:${string}`;
+
+/**
+ * Minimal ABI for BeefyVaultV7 contract
+ * @see https://github.com/beefyfinance/beefy-contracts/blob/master/contracts/BIFI/vaults/BeefyVaultV7.sol
+ */
+const BEEFY_VAULT_ABI = [
+  // Get user's vault share balance (mooToken balance)
+  'function balanceOf(address account) external view returns (uint256)',
+  // Get the current price per full share (18 decimals)
+  'function getPricePerFullShare() external view returns (uint256)',
+  // Alternative: some vaults use balance() for total underlying
+  'function balance() external view returns (uint256)',
+  // Total supply of vault shares
+  'function totalSupply() external view returns (uint256)',
+];
+
+interface BeefyBalanceClientPowers {
+  evmProviders: EvmProviders;
+  clusterName: ClusterName;
+  log?: (...args: unknown[]) => void;
+}
+
+export class BeefyBalanceClient {
+  private readonly evmProviders: EvmProviders;
+
+  private readonly clusterName: ClusterName;
+
+  private readonly log: (...args: unknown[]) => void;
+
+  constructor(powers: BeefyBalanceClientPowers) {
+    this.evmProviders = powers.evmProviders;
+    this.clusterName = powers.clusterName;
+    this.log = powers.log ?? (() => {});
+  }
+
+  /**
+   * Parse account ID to extract CAIP chain ID and address
+   * AccountId format: namespace:reference:address (e.g., "eip155:1:0xabc...")
+   */
+  private static parseAccountId(accountId: AccountId): {
```

Comment:
The `@agoric/orchestration` package provides this function, no? Any particular reason not to import it? Add a comment?